onnx/mapping.py

```diff
@@ -1,84 +1,167 @@
+    ),
+    int(TensorProto.INT8): TensorDtypeMap(
+        np.dtype("int8"), int(TensorProto.INT32), "TensorProto.INT8"
+    ),
+    int(TensorProto.UINT16): TensorDtypeMap(
+        np.dtype("uint16"), int(TensorProto.INT32), "TensorProto.UINT16"
+    ),
+    int(TensorProto.INT16): TensorDtypeMap(
+        np.dtype("int16"), int(TensorProto.INT32), "TensorProto.INT16"
+    ),
+    int(TensorProto.INT32): TensorDtypeMap(
+        np.dtype("int32"), int(TensorProto.INT32), "TensorProto.INT32"
+    ),
+    int(TensorProto.INT64): TensorDtypeMap(
+        np.dtype("int64"), int(TensorProto.INT64), "TensorProto.INT64"
+    ),
+    int(TensorProto.BOOL): TensorDtypeMap(
+        np.dtype("bool"), int(TensorProto.INT32), "TensorProto.BOOL"
+    ),
+    int(TensorProto.FLOAT16): TensorDtypeMap(
+        np.dtype("float16"), int(TensorProto.UINT16), "TensorProto.FLOAT16"
+    ),
+    # Native numpy does not support bfloat16 so now use float32 for bf16 values
+    # TODO ONNX should dirtectly use bfloat16 for bf16 values after numpy has supported bfloat16 type
+    int(TensorProto.BFLOAT16): TensorDtypeMap(
+        np.dtype("float32"), int(TensorProto.UINT16), "TensorProto.BFLOAT16"
+    ),
+    int(TensorProto.DOUBLE): TensorDtypeMap(
+        np.dtype("float64"), int(TensorProto.DOUBLE), "TensorProto.DOUBLE"
+    ),
+    int(TensorProto.COMPLEX64): TensorDtypeMap(
+        np.dtype("complex64"), int(TensorProto.FLOAT), "TensorProto.COMPLEX64"
+    ),
+    int(TensorProto.COMPLEX128): TensorDtypeMap(
+        np.dtype("complex128"), int(TensorProto.DOUBLE), "TensorProto.COMPLEX128"
+    ),
+    int(TensorProto.UINT32): TensorDtypeMap(
+        np.dtype("uint32"), int(TensorProto.UINT32), "TensorProto.UINT32"
+    ),
+    int(TensorProto.UINT64): TensorDtypeMap(
+        np.dtype("uint64"), int(TensorProto.UINT64), "TensorProto.UINT64"
+    ),
+    int(TensorProto.STRING): TensorDtypeMap(
+        np.dtype("object"), int(TensorProto.STRING), "TensorProto.STRING"
+    ),
 }
 
+
+class DeprecatedWarningDict(dict):  # type: ignore
+    def __init__(
+        self,
+        dictionary: Dict[int, Union[int, str]],
+        original_function: str,
+        future_function: str,
+    ) -> None:
+        super().__init__(dictionary)
+        self._origin_function = original_function
+        self._future_function = future_function
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, DeprecatedWarningDict):
+            return False
+        return (
+            self._origin_function == other._origin_function
+            and self._future_function == other._future_function
+        )
+
+    def __getitem__(self, key: Union[int, str]) -> Any:
+        warnings.warn(
+            str(
+                f"`mapping.{self._origin_function}` is deprecated in ONNX 1.13 and will "
+                + "be removed in next release. To silence this warning, please use `helper.{_self.future_function}` instead."
```

Comment:
Are we definitely planning to remove in release 1.14? If we are not sure, we can change the message to "will be removed soon" or "will be removed in the next release or so."

Reply:
To be flexible, I changed it into: `XXX is now deprecated and will be removed in the next release or so.`